odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Cone beam related geometries."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from math import cos, sin
+from abc import ABCMeta
+from future import standard_library
+from future.utils import with_metaclass
+standard_library.install_aliases()
+from builtins import super
+
+
+# External
+import numpy as np
+
+# Internal
+from odl.set.domain import IntervalProd
+from odl.discr.grid import TensorGrid
+from odl.tomo.geometry.detector import Flat2dDetector
+from odl.tomo.geometry.geometry import Geometry
+# from odltomo.util.trafos import euler_matrix
+
+
+__all__ = ('CircularConeFlatGeometry', 'HelicalConeFlatGeometry')
+
+
+class ConeBeamGeometry(with_metaclass(ABCMeta, Geometry)):
+
+    """Abstract 3d cone beam geometry.
+
+    The source moves on a circle with radius ``r``, and the detector
+    reference point is opposite to the source on a circle with radius ``R``.
+
+    The motion parameter is the (1d) rotation angle parametrizing source and
+    detector positions.
```

Comment:
This is good for a class docstring, but we're gonna need some "proper" documenation of this in the guide later (not for this pull, but make an issue).